tutorials/pages/shimmerevm-hardhat.md

```diff
@@ -0,0 +1,222 @@
+
+## Set Up the Contract
+
+1. Go to `hardhat.config.js`
+2. Update the `hardhat-config` with shimmer network details as shown below:
+
+```js
+module.exports = {
+  solidity: '0.8.18',
+  defaultNetwork: 'shimmerevm-testnet',
+  networks: {
+    'shimmerevm-testnet': {
+      url: 'https://json-rpc.evm.testnet.shimmer.network',
+      chainId: 1071,
+      accounts: [priv_key],
+    },
+  },
+};
+```
+
+:::info Private Key
+
+How to export the MetaMask private key of an account?
+
+- Click on the logo in the upper right corner
+
+- Select the account you want to export
+
+- On the account page, click the menu (three dots) in the upper right corner, then click the "Account Details" button:
+
+- Click "Export Private Key"
+
+- To access your private key, you will now need to enter your wallet password. When done, click Confirm to continue.
+
+- Your private key will now be displayed. Click to copy it and save it in a safe place. (Note: For obvious reasons, we didn't show it in the screenshots below - but yours will be there.)
+
+- Click close icon on top to close the screen.
+
+Follow [this article](https://helpwithpenny.com/export-and-import-metamask-private-key/) for further detail.
+
+:::
+
+1. Create a `.env` file in the project root to store your network id and jsonrpc if you want to keep it private. You can then refer to them by `process.env.SHIMMEREVM_TESTNET_JSONRPC`.
+
+:::note DOTENV
+
+Note that the file above requires DOTENV to manage environment variables and ethers and etherscan. Make sure to install all those packages.
+
+Find more instructions on how to use DOTENV in the [DOTENV npm package](https://www.npmjs.com/package/dotenv).
+
+:::
+
+## Compile the Contract
+
+To compile the contract, you first need to install the [Hardhat Toolbox package](https://www.npmjs.com/package/@nomicfoundation/hardhat-toolbox):
+
+```sh
+npm install --save-dev @nomicfoundation/hardhat-toolbox
+```
+
+Then, simply run to compile:
+
+```sh
+npx hardhat compile
+```
+
+## Deploy Your Contract on ShimmerEVM
+
+Run this command in root of the project directory:
+
+```sh
+npx hardhat run scripts/deploy.js --network shimmerevm-testnet
+```
+
+The contract will be deployed on ShimmerEVM Testnet, and you can check the deployment status here on the [explorer](https://explorer.evm.testnet.shimmer.network/).
+
+If you want to further verify your contract, add the following to your `hardhat.config.js`:
+
+```js
+etherscan: {
+  customChains: [
+    {
+      network: 'shimmerevm-testnet',
+      chainId: 1071,
+      apiKey: 'ABCDE12345ABCDE12345ABCDE123456789',
+      urls: {
+        apiURL: 'https://api.evm.testnet.shimmer.network/',
+        browserURL: 'https://explorer.evm.testnet.shimmer.network/',
+      },
+    },
+  ];
+}
+```
+
+Then you can verify by running:
+
+```sh
+npx hardhat verify --network shimmerevm-testnet <address> <unlock time>
```

Comment:
Here I use the contract address and the timestamp I got from the deploy step, correct?
I get this error:
```
Error in plugin @nomiclabs/hardhat-etherscan: Failed to send contract verification request.
Endpoint URL: https://api.evm.testnet.shimmer.network/
The HTTP server response is not ok. Status code: 404 Response text: {"Message":"Not Found","Error":""}


NomicLabsHardhatPluginError: Failed to send contract verification request.
Endpoint URL: https://api.evm.testnet.shimmer.network/
The HTTP server response is not ok. Status code: 404 Response text: {"Message":"Not Found","Error":""}
```

So is something still wrong with the config?

Reply:
Yeah getting error now. Maybe api is down. Had tested this a while ago. Let me get back on this.

Reply:
My bad. The URL was wrong. Try now. Should work.

Reply:
Hmmm. Maybe I'm too stupid for this, but now I get:
```
Nothing to compile
Successfully submitted source code for contract
contracts/Lock.sol:Lock at 0x30a773e646f984f23b5673b2fde78C18b4B0FB01
for verification on the block explorer. Waiting for verification result...

We tried verifying your contract Lock without including any unrelated one, but it failed.
Trying again with the full solc input used to compile and deploy it.
This means that unrelated contracts may be displayed on Etherscan...

Successfully submitted source code for contract
contracts/Lock.sol:Lock at 0x30a773e646f984f23b5673b2fde78C18b4B0FB01
for verification on the block explorer. Waiting for verification result...

Error in plugin @nomiclabs/hardhat-etherscan: The API responded with an unexpected message.
Contract verification may have succeeded and should be checked manually.
Message: Unknown UID

NomicLabsHardhatPluginError: The API responded with an unexpected message.
Contract verification may have succeeded and should be checked manually.
Message: Unknown UID
```

Reply:
Ok, seems to be a network problem. Worked now without problems